shmi_darwin.go

```diff
@@ -0,0 +1,148 @@
+// +build darwin,cgo
+
+package shm
+
+/*
+#include <sys/mman.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <stdio.h>
+#include <unistd.h>
+
+int _create(const char* name, int size, int flag) {
+	mode_t mode = S_IRUSR | S_IWUSR | S_IRGRP | S_IWGRP;
+	int fd = shm_open(name, flag, mode);
+	if (fd < 0) {
+		return -1;
+	}
+	struct stat mapstat;
+	if (-1 != fstat(fd, &mapstat) && mapstat.st_size == 0) {
+		if (ftruncate(fd, size) != 0) {
+		close(fd);
+		return -2;
+		}
+	}
+	return fd;
```

Comment:
What happens if `fstat` returns `-1` ?

Reply:
it should return an error

Reply:
it should closed file and return error

Reply:
This is done because the shm_open api for mac is slightly different from linux.